platforms/chibios/drivers/ws2812_pwm.c

```diff
@@ -247,13 +263,33 @@
 
 /* --- PRIVATE VARIABLES ---------------------------------------------------- */
 
+// STM32F2XX, STM32F4XX and STM32F7XX do NOT zero pad DMA transfers of unequal data width. Buffer width must match TIMx CCR.
+// For all other STM32 DMA transfer will automatically zero pad. We only need to set the right peripheral width.
+#if defined(STM32F2XX) || defined(STM32F4XX) || defined(STM32F7XX)
+#    if defined(WS2812_PWM_TIMER_32BIT)
+#        define WS2812_DMA_MEMORY_WIDTH STM32_DMA_CR_MSIZE_WORD
+#        define WS2812_DMA_PERIPHERAL_WIDTH STM32_DMA_CR_PSIZE_WORD
 static uint32_t ws2812_frame_buffer[WS2812_BIT_N + 1]; /**< Buffer for a frame */
```

Comment:
Maybe add a typedef or something, and then define `ws2812_frame_buffer` only once?